### PR TITLE
upd readme [duda]

### PR DESCRIPTION
**Pull request by Dmitry-Duda**

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -3,20 +3,26 @@
...
 
-Starting from version 17.1 we have introduced a new client-side <a href="https://documentation.devexpress.com/#AspNet/DevExpressWebScriptsASPxClientGridView_FocusedCellChangingtopic">FocusedCellChanging</a> event. This event allows you to skip focusing and editing a cell. <br>This example demonstrates different ways to skip or prevent editing certain cells.<br>The main idea is to get the currently focused cell using the <a href="https://documentation.devexpress.com/#AspNet/DevExpressWebScriptsASPxClientGridViewFocusedCellChangingEventArgs_cellInfotopic">ASPxClientGridViewFocusedCellChangingEventArgs.cellInfo</a> property and decide whether it is required to skip focusing or not:<br>
 
+## Implementation Details
+
+On the client side, the [`FocusedCellChanging`](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.FocusedCellChanging) event contains clicked cell data of the [`ASPxClientGridViewCellInfo`](https://docs.devexpress.com/AspNet/js-ASPxClientGridViewCellInfo) class. To get this data, the code below uses the event handler argument's [`cellInfo`](https://docs.devexpress.com/AspNet/js-ASPxClientGridViewFocusedCellChangingEventArgs.cellInfo) property.
```
Comment (alexey-zolotarev) [inline Readme.md]:
```suggestion
On the client side, the [`FocusedCellChanging`](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.FocusedCellChanging) event contains the [`cellInfo`](https://docs.devexpress.com/AspNet/js-ASPxClientGridViewFocusedCellChangingEventArgs.cellInfo) event argument that specifies the clicked cell's data (an [`ASPxClientGridViewCellInfo`](https://docs.devexpress.com/AspNet/js-ASPxClientGridViewCellInfo) object).
```

File: Readme.md [hunk 2]
```diff
@@ -3,20 +3,26 @@
...
+
+Depending on the obtained data, the handler code determines whether a user can focus and then edit the clicked cell. For this purpose, the code specifies the handler argument's [`cancel`](https://docs.devexpress.com/AspNet/js-ASPxClientCancelEventArgs.cancel) property. When the [`cancel`](https://docs.devexpress.com/AspNet/js-ASPxClientCancelEventArgs.cancel) property is **true**, the grid cancels the focus action for the clicked cell.
```
Comment (alexey-zolotarev) [inline Readme.md]:
```suggestion
The event handler reads the cell's data to determine whether a user can focus and then edit the clicked cell. Set the [`cancel`](https://docs.devexpress.com/AspNet/js-ASPxClientCancelEventArgs.cancel) event property to **true** to cancel the focus action for the clicked cell.
```

File: Readme.md [hunk 3]
```diff
@@ -31,9 +37,4 @@ function onFocusedCellChanging(s, e) {
 }
 ```
 
-
-<p><strong>Note</strong>: the FocusedCellChanging event can't prevent editor focusing if the <a href="https://documentation.devexpress.com/#AspNet/DevExpressWebASPxGridView_SettingsEditingtopic">ASPxGridView.SettingsEditing</a> -> <a href="https://documentation.devexpress.com/#AspNet/DevExpressWebASPxGridViewEditingSettings_BatchEditSettingstopic">BatchEditSettings</a> -> <a href="https://documentation.devexpress.com/#AspNet/DevExpressWebGridViewBatchEditSettings_EditModetopic">EditMode</a> property is set to Row.</p>
-
-<br/>
-
-
+Note that this technique does not work if you set the [`EditMode`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewBatchEditSettings.EditMode) property to **Row**. A user can focus and then edit any cell in the row switched to edit mode (except for cells of columns with the enabled [`ReadOnly`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewDataColumn.ReadOnly) property).
```
Comment (alexey-zolotarev) [inline Readme.md]:
```suggestion
This technique is not applicable if you set the [`EditMode`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewBatchEditSettings.EditMode) property to **Row**. A user can focus and edit any cell in the row switched to edit mode except for read-only column cells (see [`GridViewDataColumn.ReadOnly`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewDataColumn.ReadOnly)).
```

File: Readme.md (hunk 1 above)
Comment (Dmitry-Duda) [reply Readme.md]:
```suggestion
The following client-side [`FocusedCellChanging`] event handler uses the [`e.Cancel`] property to cancel the focus action (and subsequent edit operations) for cells in specific columns and rows. The code uses the [`e.cellInfo`] event property to get information on the clicked cell's column and row.
```

File: Readme.md (hunk 2 above)
Comment (Dmitry-Duda) [reply Readme.md]:
```suggestion
```

File: Readme.md (hunk 3 above)
Comment (Dmitry-Duda) [reply Readme.md]:
```suggestion
This technique is not applicable if you set the [`EditMode`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewBatchEditSettings.EditMode) property to **Row**. A user can focus and edit any cell in the row switched to edit mode except for cells in read-only columns (the [`GridViewDataColumn.ReadOnly`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewDataColumn.ReadOnly) property).
```

File: Readme.md [hunk 4]
```diff
@@ -31,9 +36,4 @@ function onFocusedCellChanging(s, e) {
 }
 ```
 
-
-<p><strong>Note</strong>: the FocusedCellChanging event can't prevent editor focusing if the <a href="https://documentation.devexpress.com/#AspNet/DevExpressWebASPxGridView_SettingsEditingtopic">ASPxGridView.SettingsEditing</a> -> <a href="https://documentation.devexpress.com/#AspNet/DevExpressWebASPxGridViewEditingSettings_BatchEditSettingstopic">BatchEditSettings</a> -> <a href="https://documentation.devexpress.com/#AspNet/DevExpressWebGridViewBatchEditSettings_EditModetopic">EditMode</a> property is set to Row.</p>
-
-<br/>
-
-
+This technique is not applicable if you set the [`EditMode`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewBatchEditSettings.EditMode) property to **Row**. A user can focus and edit any cell in the row switched to edit mode except for cells in read-only columns (the [`GridViewDataColumn.ReadOnly`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewDataColumn.ReadOnly) property).
```
Comment (dirk-pieterse) [inline Readme.md:39]:
```suggestion
This technique is not applicable if you set the [`EditMode`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewBatchEditSettings.EditMode) property to **Row**. A user can focus and edit any cell in rows switched to edit mode except for cells in read-only columns (the [`GridViewDataColumn.ReadOnly`](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewDataColumn.ReadOnly) property).
```

**Review by dirk-pieterse — APPROVED** [review]

File: Readme.md (hunk 4 above)
Comment (Dmitry-Duda) [reply Readme.md:39]: As a result of the discussion, we've decided to keep the "the row".

**Pull request merged**